components/internal/ApplicantStatusBadge.tsx

```diff
@@ -0,0 +1,118 @@
+import { Badge, Box } from '@chakra-ui/react'; // Chakra UI
+import Image from 'next/image'; // Optimized images
+
+type Props = {
+  variant: string;
+};
+
+/**
+ * Renders the applicant status badge content (icon + text)
+ * @param variant for specifying applicant status badge type
+ * @returns {JSX.Element} contents of the badge
+ */
+const renderBadgeContent = (variant: string): JSX.Element | undefined => {
```

Comment:
We don't want to render `undefined` - we can work around this with the suggestion I posted above. Then, we can replace the `switch` statement with an `if-else` statement and cover all of the possible values. TypeScript shouldn't complain if we do that.

Reply:
I tried that out and it still complains unless I have a final `else` statement but that's the equivalent of specifying the `default` case in a switch statement. I opted to still use the switch statment but made the `active` case the default case so that fixes it. Lmk if you have any issues with that!

Code snippet:
```ts
case 'active':
      default:
        return (
....
```

Reply:
That is fine!